datatypes/src/raster/no_data_grid.rs

```diff
@@ -0,0 +1,172 @@
+use std::ops::Add;
+
+use super::{
+    grid_traits::{ChangeGridBounds, GridShapeAccess},
+    Grid, GridBoundingBox, GridBounds, GridIdx, GridIndexAccess, GridShape, GridShape1D,
+    GridShape2D, GridShape3D, GridSize, GridSpaceToLinearSpace, NoDataValue,
+};
+use crate::{
+    error::{self},
+    raster::GridContains,
+    util::Result,
+};
+use num_traits::AsPrimitive;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NoDataGrid<D, T> {
+    pub shape: D,
+    pub no_data_value: T,
+}
+
+pub type NoDataGrid1D<T> = NoDataGrid<GridShape1D, T>;
+pub type NoDataGrid2D<T> = NoDataGrid<GridShape2D, T>;
+pub type NoDataGrid3D<T> = NoDataGrid<GridShape3D, T>;
+
+impl<D, T> NoDataGrid<D, T>
+where
+    D: GridSize,
+    T: Clone,
+{
+    /// Creates a new `Grid`
+    ///
+    /// # Errors
+    ///
+    /// This constructor fails if the data container's capacity is different from the grid's dimension number
+    ///
+    pub fn new(shape: D, no_data_value: T) -> Result<Self> {
+        Ok(Self {
+            shape,
+            no_data_value,
+        })
+    }
+
+    /// Converts the data type of the raster by converting it pixel-wise
+    pub fn convert_dtype<To>(self) -> NoDataGrid<D, To>
+    where
+        T: AsPrimitive<To> + Copy + 'static,
+        To: Copy + 'static,
+    {
+        NoDataGrid::new(self.shape, self.no_data_value.as_())
+            .expect("grid array type conversion cannot fail")
```

Comment:
Why don't you just construct the type without using the constructor?

Reply:
no longer a problem since it cant fail